### PR TITLE
Remove some older EVM recursion logic

### PR DESCRIPTION
**Pull request by dlubarov**
Some logic was replaced by the constant-degree logic in `fixed_recursive_verifier`.

### CODE REVIEW

File: evm/src/proof.rs [hunk 1]
```diff
@@ -46,11 +46,6 @@ pub(crate) struct AllChallengerState<F: RichField + Extendable<D>, const D: usiz
     pub ctl_challenges: GrandProductChallengeSet<F>,
 }
 
-pub struct AllProofTarget<const D: usize> {
```
Comment (dlubarov) [inline evm/src/proof.rs:49]: (We no longer have one circuit that checks all STARK proofs)

File: evm/src/recursive_verifier.rs [hunk 2]
```diff
@@ -59,11 +49,6 @@ pub struct RecursiveAllProof<
     pub recursive_proofs: [ProofWithPublicInputs<F, C, D>; NUM_TABLES],
 }
 
-pub struct RecursiveAllProofTargetWithData<const D: usize> {
```
Comment (dlubarov) [inline evm/src/recursive_verifier.rs:62]: (`AllRecursiveCircuits::by_table` now holds this data)

File: evm/src/recursive_verifier.rs [hunk 3]
```diff
@@ -427,85 +334,6 @@ pub(crate) fn add_common_recursion_gates<F: RichField + Extendable<D>, const D:
     )));
 }
 
-/// Returns the recursive Stark circuit verifier data for every Stark in `AllStark`.
-pub fn all_verifier_data_recursive_stark_proof<
```
Comment (dlubarov) [inline evm/src/recursive_verifier.rs:431]: (This logic is now in `AllRecursiveCircuits::new` and its helpers)

**Review by wborgeaud — APPROVED** [review]
Looks good to me!

**Pull request merged**